iceoryx_utils/include/iceoryx_utils/cxx/uninitialized_array.hpp

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) 2021 by Robert Bosch GmbH. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+#ifndef ICEORYX_UTILS_CXX_UNINITIALIZED_ARRAY_HPP
+#define ICEORYX_UTILS_CXX_UNINITIALIZED_ARRAY_HPP
+
+#include <cstdint>
+
+namespace iox
+{
+namespace cxx
+{
+template <typename T, uint64_t Capacity>
+class uninitialized_array
+{
+  public:
+    using value_type = T;
+    using size_type = decltype(Capacity);
+
+    uninitialized_array() = default;
+
```

Comment:
This thing should be neither move nor copyable otherwise we will run into serious troubles.
Could you please delete copy and move ctor/assignment.

This is just a building block and the copy/move has to be implemented from the class which is using this as a member.

Reply:
Agreed.